crates/sui-json-rpc/src/api/write.rs

```diff
@@ -40,9 +40,16 @@ pub trait WriteApi {
         request_type: Option<ExecuteTransactionRequestType>,
     ) -> RpcResult<SuiTransactionBlockResponse>;
 
-    /// Runs the transaction in dev-inspect mode. Which allows for nearly any
-    /// transaction (or Move call) with any arguments. Detailed results are
-    /// provided, including both the transaction effects and any return values.
+    /// Simulates running the transaction, even one that would be normally
+    /// impossible, against the current state of the network. Nearly any
+    /// transaction (or Move call) with any arguments can run in dev-inspect;
+    /// transaction inputs are not checked, entry function verification is
+    /// skipped, values can be left unused, and returns are not validated.
+    /// Detailed results are provided, including both the transaction effects
+    /// and any return values. Dev-inspect is provided to aid in understanding
+    /// the effects of any possible action against the current network state.
+    /// See dryRunTransactionBlock for a more faithful simulation of what would
+    /// happen if the transaction were actually run.
```

Comment:
```suggestion
    /// Simulates running the transaction against the network, even if 
    /// the transaction is not valid. You can use `dev_Inspect` mode to 
    /// simulate any transaction, or Sui Move call, and provide any
    /// argument or value in the request. When you run `devInspect`, it 
    /// doesn't check for valid inputs, verify entry functions, or validate 
    /// transaction results. You can also run requests without
    /// specifying argument values at all. These requests return the result 
    /// of the simulated transaction effects. Use `dev_Inspect` to identify the 
    /// potential outcome of a transaction against a Sui network.
    /// For a more accurate simulation of actually running a transaction, use 
    /// `dryRunTransactionBlock`.
```

I don't understand the delta between the 2 - simulates a transaction, but for a more faithful view use this other thing. I am left wondering which to use and why.

Reply:
Clarified some of the explanations of the restrictions that were loosened and tried to distance `devInspect` and `dryRun` even more by saying the `dryRun` is for "simulation" and `devInspect` is to "inspect the state".

```suggestion
    /// Inspect the current state of the network by running a programmable
    /// transaction block without committing its effects on chain.  Unlike 
    /// `dryRunTransactionBlock`, dev inspect will not validate whether the 
    /// transaction block would succeed or fail under normal circumstances,
    /// e.g.:
    ///
    /// - Transaction inputs are not checked for ownership (i.e. you can
    ///   construct calls involving objects you do not own).
    /// - Calls are not checked for visibility (you can call private functions 
    ///    on modules)
    /// - Inputs of any type can be constructed and passed in, (including 
    ///    Coins and other objects that would usually need to be constructed
    ///    with a move call).
    /// - Function returns do not need to be used, even if they do not have 
    ///   `drop`.
    ///
    /// Dev inspect's output includes a breakdown of results returned by 
    /// every transaction in the block, as well as the transaction's effects.
    ///
    /// To run an accurate simulation of a transaction and understand whether
    /// it will successfully validate and run, use `dryRunTransactionBlock` 
    /// instead.
```